solutions/devsprint-riccardo-1/OnboardingChallenge/Screens/Settings/SettingsInteractor.swift

```diff
@@ -1,28 +1,21 @@
-//
-//  SettingsInteractor.swift
-//  OnboardingChallenge
-//
-//  Created by Strawberry Pie on 12/14/21.
-//
+// Copyright © 2021 Bending Spoons S.p.A. All rights reserved.
 
 import Foundation
 
-protocol SettingsBusinessLogic {
-    func getAppVersion(request: SettingsViewConfiguration.Get.Request)
+protocol LoadingPropertyInfo {
+  func getAppVersion(completion: @escaping (SettingsViewModel) -> Void)
 }
 
-class SettingsInteractor: SettingsBusinessLogic {
-    var presenter: SettingsPresentationLogic?
-    var bundle: BundleService?
-    
-    init(bundle: BundleService) {
-        self.bundle = bundle
-    }
-    
-    func getAppVersion(request: SettingsViewConfiguration.Get.Request) {
-        guard let value = self.bundle?.object(forInfoDictionaryKey: request.key) as? String else { return }
-        let response = SettingsViewConfiguration.Get.Response(item: value)
-        
-        self.presenter?.presenter(response: response)
-    }
+class SettingsInteractor: LoadingPropertyInfo {
+  var bundleService: BundleService?
```

Comment:
why is this optional? We are passing it to the `init` directly, so we can remove the `?`. This simplifies also the implementation of the `GetAppVersion`directly. 

I'll also make it a `let` instead of a `var` property: we don't want that something from the outside changes the value of the `bundleService` while the `ViewController` is using it.